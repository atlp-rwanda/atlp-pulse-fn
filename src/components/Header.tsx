import React, { forwardRef, useContext } from 'react';
import { useTranslation } from 'react-i18next';
import { Link, NavLink , useLocation} from 'react-router-dom';
import { MenuIcon, XIcon } from '@heroicons/react/outline';
import { UserContext } from '../hook/useAuth';
import Button from './Buttons';
import WithClickOutside from './WithClickOutside';
import LogoIcon from './logoIcon';
import ToggleThemeButton from './TogglethemeIcon';

const Header = forwardRef(({ open, setOpen, ...props }: any, ref: any) => {
  const orgToken: any = localStorage.getItem('orgToken');
  const { t } = useTranslation();

  /* istanbul ignore next */
  const handleClick = () => setOpen(!open);
  const { user, logout } = useContext(UserContext);
  const location = useLocation()
  const pathname = location.pathname.split("/")[1]
  
  const goTo = orgToken ? '/users/login' : '/login/org';

  return (
    <div
      className={`w-screen h-[8vh] z-10 bg-white dark:bg-dark-bg page-header ${props?.className}`}
      style={props?.style}
    >
      <div className="px-3 flex justify-between items-center w-full h-full">
        <div className="flex items-center h-full justify-between lg:w-full">
          <Link
            to="/"
            className="flex flex-row lg:px-5 text-primary dark:text-dark-text-fill"
          >
            <LogoIcon />
            <h1 className="text-3xl font-bold font-lexend">PULSE</h1>
          </Link>
          <ul className="hidden cursor-pointer lg:flex">
            <li className="px-5 text-xl dark:text-dark-text-fill">
              <NavLink
                to="/"
                className={(navData) => {
                  if (navData.isActive) return 'text-primary';
                  return '';
                }}
              >
                {t('Home')}
              </NavLink>
            </li>
            <li className="px-5 text-xl dark:text-dark-text-fill">
              <NavLink
                className={(navData) => {
                  if (navData.isActive) return 'text-primary';
                  return '';
                }}
                to="/about"
              >
                {t('About')}
              </NavLink>
            </li>
            {!user?.auth ? (
              <li className="px-5 text-xl dark:text-dark-text-fill">
                <NavLink
                  className={(navData) => {
                    if (navData.isActive) return 'text-primary';
                    return '';
                  }}
                  to="/pricing"
                >
                  {t('Pricing')}
                </NavLink>
              </li>
            ) : (
              ' '
            )}
            <li className="px-5 text-xl dark:text-dark-text-fill">
              <NavLink
                className={() => {
                  if (pathname==="docs") return 'text-primary';
                  return '';
                }}
                to="/docs/org-signin"
              >
                {t('Docs')}
              </NavLink>
            </li>
          </ul>
        </div>
        <div className="hidden lg:flex w-2/3 justify-end ">
          <ToggleThemeButton />
          <Link to={user?.auth ? '/dashboard' : goTo}>
            <Button variant="primary" size="lg">
              {' '}
              {!user?.auth ? t('Sign In') : t('Dashboard')}{' '}
            </Button>
          </Link>
          {user?.auth ? (
            <Button
              variant="transparentbtn"
              size="lg"
              onClick={() => logout()}
              style="text-red-500 font-bolf dark:text-dark-text-fill mr-8 border border-red-600 dark:border-dark-text-fill"
            >
              {' '}
              {t('Logout')}{' '}
            </Button>
          ) : (
            <Link to="/signup/org">
              <Button variant="transparentbtn" size="lg" style="mr-8">
                {' '}
                {t('Register an organization')}{' '}
              </Button>
            </Link>
          )}
        </div>
        <div className="flex px-5 lg:hidden">
          <ToggleThemeButton />
          <button type="button" onClick={handleClick}>
            {!open ? (
              <MenuIcon className="w-7 dark:text-dark-text-fill" />
            ) : (
              <XIcon className="w-7 dark:text-dark-text-fill" />
            )}
          </button>
        </div>
      </div>
      <ul
        ref={ref}
        className={
          !open
            ? 'hidden'
            : 'absolute bg-white dark:bg-dark-bg w-1/8 justify-end px-8 m-1 right-0 lg:hidden'
        }
      >
        <li className="w-full p-2 mt-2 dark:text-dark-text-fill text-primary">
          <Link to="/">{t('Home')}</Link>
        </li>
        {!user?.auth ? (
          <li className="w-full p-2 dark:text-dark-text-fill">
            <Link to="/pricing">Pricing</Link>
          </li>
        ) : (
          ' '
        )}

        <li className="w-full p-2 dark:text-dark-text-fill">
          <Link to="/product" className="w-full">
            {t('Product')}
          </Link>
        </li>
        <li className="p-2 w-full mt-2 dark:text-dark-text-fill text-primary">
          <Link to="/docs/org-signin">{t('Docs')}</Link>
        </li>

        <li className="p-2 w-full dark:text-dark-text-fill mt-6 mb-2 bg-primary text-white rounded-md px-[35%]">
          <Link to={user?.auth ? '/dashboard' : goTo} className="w-full">
            {' '}
            {!user?.auth ? t('Sign In') : t('Dashboard')}
          </Link>
        </li>
        {user?.auth ? (
          <Button
            variant="transparentbtn"
            size="lg"
            onClick={() => logout()}
            style="text-red-500 font-bolf dark:text-dark-text-fill mr-8 border border-red-600 dark:border-dark-text-fill"
          >
            {t('Logout')}
          </Button>
        ) : (
          <Link to="/signupOrg">
            <Button variant="transparentbtn" size="lg" style="mr-8">
              {' '}
              {t('Register an organization')}{' '}
            </Button>
          </Link>
        )}
      </ul>
    </div>
  );
});

export default WithClickOutside(Header);
