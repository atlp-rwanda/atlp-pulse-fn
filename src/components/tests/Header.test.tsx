/* eslint-disable */
import * as React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import UserProvider, { UserContext } from '../../hook/useAuth';
import Header from '../Header';

describe('Header Tests', () => {
  it('Should render the header with pricing active', () => {
    const elem = renderer
      .create(
        <MemoryRouter initialEntries={['/pricing']}>
          <Header></Header>
        </MemoryRouter>,
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
  it('Should render the header with home', () => {
    const elem = renderer
      .create(
        <MemoryRouter initialEntries={['/']}>
          <Header></Header>
        </MemoryRouter>,
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
  it('Should render the header with product active', () => {
    const elem = renderer
      .create(
        <MemoryRouter initialEntries={['/product']}>
          <Header></Header>
        </MemoryRouter>,
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
  it('Should render the header with no active', () => {
    const elem = renderer
      .create(
        <MemoryRouter initialEntries={['/no-active']}>
          <Header></Header>
        </MemoryRouter>,
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
  it('Should render the header with no active', () => {
    const elem = renderer
      .create(
        <UserContext.Provider
          value={{ user: { auth: true, name: 'name', role: 'admin' } }}
        >
          <MemoryRouter initialEntries={['/no-active']}>
            <Header></Header>
          </MemoryRouter>
        </UserContext.Provider>,
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
  it('Should render the header with no active', () => {
    const elem = renderer
      .create(
        <UserProvider>
          <MemoryRouter initialEntries={['/']}>
            <Header></Header>
          </MemoryRouter>
        </UserProvider>,
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
});
