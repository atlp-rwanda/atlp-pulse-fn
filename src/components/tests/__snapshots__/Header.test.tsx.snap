// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Tests Should render the header with home 1`] = `
<div
  className="w-screen h-[8vh] z-10 bg-white dark:bg-dark-bg fixed border-b undefined"
>
  <div
    className="px-3 flex justify-between items-center w-full h-full"
  >
    <div
      className="flex items-center h-full justify-between lg:w-full"
    >
      <a
        className="flex flex-row lg:px-5"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo"
          className="w-full cursor-pointer mr-2"
          src={Object {}}
        />
        <h1
          className="text-3xl font-bold font-lexend text-primary dark:text-dark-text-fill"
        >
          PULSE
        </h1>
      </a>
      <ul
        className="hidden lg:flex cursor-pointer"
      >
        <li
          className="px-5 text-xl dark:text-dark-text-fill"
        >
          <a
            aria-current="page"
            className="text-primary"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="px-5 text-xl dark:text-dark-text-fill"
        >
          <a
            className=""
            href="/pricing"
            onClick={[Function]}
          >
            Pricing
          </a>
        </li>
        <li
          className="px-5 text-xl dark:text-dark-text-fill"
        >
          <a
            className=""
            href="/product"
            onClick={[Function]}
          >
            Product
          </a>
        </li>
      </ul>
    </div>
    <div
      className="hidden lg:flex lg:w-full justify-end "
    >
      <button
        className="px-4 mt-1 cursor-pointer"
        id="theme-switch"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="w-8"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17.293 13.293A8 8 0 016.707 2.707a8.001 8.001 0 1010.586 10.586z"
          />
        </svg>
      </button>
      <a
        href="/organization-login"
        onClick={[Function]}
      >
        <button
          className="btn primary lg undefined"
          disabled={false}
          type="button"
        >
           
          Sign In
           
        </button>
      </a>
      <a
        href="/register-organization"
        onClick={[Function]}
      >
        <button
          className="btn transparentbtn lg mr-8"
          disabled={false}
          type="button"
        >
           
          Register an organization
           
        </button>
      </a>
    </div>
    <div
      className="flex px-5 lg:hidden"
    >
      <button
        className="px-3"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="w-6 mr-2"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17.293 13.293A8 8 0 016.707 2.707a8.001 8.001 0 1010.586 10.586z"
          />
        </svg>
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="w-7 dark:text-dark-text-fill"
          fill="none"
          stroke="currentColor"
          strokeWidth={2}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 6h16M4 12h16M4 18h16"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </button>
    </div>
  </div>
  <ul
    className="hidden"
  >
    <li
      className="p-2 w-full mt-2 dark:text-dark-text-fill text-primary"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="p-2 w-full dark:text-dark-text-fill"
    >
      <a
        href="/pricing"
        onClick={[Function]}
      >
        Pricing
      </a>
    </li>
    <li
      className="p-2 w-full dark:text-dark-text-fill"
    >
      <a
        className="w-full"
        href="/product"
        onClick={[Function]}
      >
        Product
      </a>
    </li>
    <li
      className="p-2 w-full dark:text-dark-text-fill mt-6 mb-2 bg-primary text-white rounded-md px-[35%]"
    >
      <a
        className="w-full"
        href="/organization-login"
        onClick={[Function]}
      >
         
        Sign In
      </a>
    </li>
    <a
      href="/register-organization"
      onClick={[Function]}
    >
      <button
        className="btn transparentbtn lg mr-8"
        disabled={false}
        type="button"
      >
         
        Register an organization
         
      </button>
    </a>
  </ul>
</div>
`;

exports[`Header Tests Should render the header with no active 1`] = `
<div
  className="w-screen h-[8vh] z-10 bg-white dark:bg-dark-bg fixed border-b undefined"
>
  <div
    className="px-3 flex justify-between items-center w-full h-full"
  >
    <div
      className="flex items-center h-full justify-between lg:w-full"
    >
      <a
        className="flex flex-row lg:px-5"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo"
          className="w-full cursor-pointer mr-2"
          src={Object {}}
        />
        <h1
          className="text-3xl font-bold font-lexend text-primary dark:text-dark-text-fill"
        >
          PULSE
        </h1>
      </a>
      <ul
        className="hidden lg:flex cursor-pointer"
      >
        <li
          className="px-5 text-xl dark:text-dark-text-fill"
        >
          <a
            className=""
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="px-5 text-xl dark:text-dark-text-fill"
        >
          <a
            className=""
            href="/pricing"
            onClick={[Function]}
          >
            Pricing
          </a>
        </li>
        <li
          className="px-5 text-xl dark:text-dark-text-fill"
        >
          <a
            className=""
            href="/product"
            onClick={[Function]}
          >
            Product
          </a>
        </li>
      </ul>
    </div>
    <div
      className="hidden lg:flex lg:w-full justify-end "
    >
      <button
        className="px-4 mt-1 cursor-pointer"
        id="theme-switch"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="w-8"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17.293 13.293A8 8 0 016.707 2.707a8.001 8.001 0 1010.586 10.586z"
          />
        </svg>
      </button>
      <a
        href="/organization-login"
        onClick={[Function]}
      >
        <button
          className="btn primary lg undefined"
          disabled={false}
          type="button"
        >
           
          Sign In
           
        </button>
      </a>
      <a
        href="/register-organization"
        onClick={[Function]}
      >
        <button
          className="btn transparentbtn lg mr-8"
          disabled={false}
          type="button"
        >
           
          Register an organization
           
        </button>
      </a>
    </div>
    <div
      className="flex px-5 lg:hidden"
    >
      <button
        className="px-3"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="w-6 mr-2"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17.293 13.293A8 8 0 016.707 2.707a8.001 8.001 0 1010.586 10.586z"
          />
        </svg>
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="w-7 dark:text-dark-text-fill"
          fill="none"
          stroke="currentColor"
          strokeWidth={2}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 6h16M4 12h16M4 18h16"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </button>
    </div>
  </div>
  <ul
    className="hidden"
  >
    <li
      className="p-2 w-full mt-2 dark:text-dark-text-fill text-primary"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="p-2 w-full dark:text-dark-text-fill"
    >
      <a
        href="/pricing"
        onClick={[Function]}
      >
        Pricing
      </a>
    </li>
    <li
      className="p-2 w-full dark:text-dark-text-fill"
    >
      <a
        className="w-full"
        href="/product"
        onClick={[Function]}
      >
        Product
      </a>
    </li>
    <li
      className="p-2 w-full dark:text-dark-text-fill mt-6 mb-2 bg-primary text-white rounded-md px-[35%]"
    >
      <a
        className="w-full"
        href="/organization-login"
        onClick={[Function]}
      >
         
        Sign In
      </a>
    </li>
    <a
      href="/register-organization"
      onClick={[Function]}
    >
      <button
        className="btn transparentbtn lg mr-8"
        disabled={false}
        type="button"
      >
         
        Register an organization
         
      </button>
    </a>
  </ul>
</div>
`;

exports[`Header Tests Should render the header with no active 2`] = `
<div
  className="w-screen h-[8vh] z-10 bg-white dark:bg-dark-bg fixed border-b undefined"
>
  <div
    className="px-3 flex justify-between items-center w-full h-full"
  >
    <div
      className="flex items-center h-full justify-between lg:w-full"
    >
      <a
        className="flex flex-row lg:px-5"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo"
          className="w-full cursor-pointer mr-2"
          src={Object {}}
        />
        <h1
          className="text-3xl font-bold font-lexend text-primary dark:text-dark-text-fill"
        >
          PULSE
        </h1>
      </a>
      <ul
        className="hidden lg:flex cursor-pointer"
      >
        <li
          className="px-5 text-xl dark:text-dark-text-fill"
        >
          <a
            className=""
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="px-5 text-xl dark:text-dark-text-fill"
        >
          <a
            className=""
            href="/pricing"
            onClick={[Function]}
          >
            Pricing
          </a>
        </li>
        <li
          className="px-5 text-xl dark:text-dark-text-fill"
        >
          <a
            className=""
            href="/product"
            onClick={[Function]}
          >
            Product
          </a>
        </li>
      </ul>
    </div>
    <div
      className="hidden lg:flex lg:w-full justify-end "
    >
      <button
        className="px-4 mt-1 cursor-pointer"
        id="theme-switch"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="w-8"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17.293 13.293A8 8 0 016.707 2.707a8.001 8.001 0 1010.586 10.586z"
          />
        </svg>
      </button>
      <a
        href="/dashboard"
        onClick={[Function]}
      >
        <button
          className="btn primary lg undefined"
          disabled={false}
          type="button"
        >
           
          Dashboard
           
        </button>
      </a>
      <button
        className="btn transparentbtn lg text-red-500 font-bolf dark:text-dark-text-fill mr-8 border border-red-600 dark:border-dark-text-fill"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
         
        Logout
         
      </button>
    </div>
    <div
      className="flex px-5 lg:hidden"
    >
      <button
        className="px-3"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="w-6 mr-2"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17.293 13.293A8 8 0 016.707 2.707a8.001 8.001 0 1010.586 10.586z"
          />
        </svg>
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="w-7 dark:text-dark-text-fill"
          fill="none"
          stroke="currentColor"
          strokeWidth={2}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 6h16M4 12h16M4 18h16"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </button>
    </div>
  </div>
  <ul
    className="hidden"
  >
    <li
      className="p-2 w-full mt-2 dark:text-dark-text-fill text-primary"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="p-2 w-full dark:text-dark-text-fill"
    >
      <a
        href="/pricing"
        onClick={[Function]}
      >
        Pricing
      </a>
    </li>
    <li
      className="p-2 w-full dark:text-dark-text-fill"
    >
      <a
        className="w-full"
        href="/product"
        onClick={[Function]}
      >
        Product
      </a>
    </li>
    <li
      className="p-2 w-full dark:text-dark-text-fill mt-6 mb-2 bg-primary text-white rounded-md px-[35%]"
    >
      <a
        className="w-full"
        href="/dashboard"
        onClick={[Function]}
      >
         
        Dashboard
      </a>
    </li>
    <button
      className="btn transparentbtn lg text-red-500 font-bolf dark:text-dark-text-fill mr-8 border border-red-600 dark:border-dark-text-fill"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
       
      Logout
       
    </button>
  </ul>
</div>
`;

exports[`Header Tests Should render the header with no active 3`] = `
<div
  className="w-screen h-[8vh] z-10 bg-white dark:bg-dark-bg fixed border-b undefined"
>
  <div
    className="px-3 flex justify-between items-center w-full h-full"
  >
    <div
      className="flex items-center h-full justify-between lg:w-full"
    >
      <a
        className="flex flex-row lg:px-5"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo"
          className="w-full cursor-pointer mr-2"
          src={Object {}}
        />
        <h1
          className="text-3xl font-bold font-lexend text-primary dark:text-dark-text-fill"
        >
          PULSE
        </h1>
      </a>
      <ul
        className="hidden lg:flex cursor-pointer"
      >
        <li
          className="px-5 text-xl dark:text-dark-text-fill"
        >
          <a
            aria-current="page"
            className="text-primary"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="px-5 text-xl dark:text-dark-text-fill"
        >
          <a
            className=""
            href="/pricing"
            onClick={[Function]}
          >
            Pricing
          </a>
        </li>
        <li
          className="px-5 text-xl dark:text-dark-text-fill"
        >
          <a
            className=""
            href="/product"
            onClick={[Function]}
          >
            Product
          </a>
        </li>
      </ul>
    </div>
    <div
      className="hidden lg:flex lg:w-full justify-end "
    >
      <button
        className="px-4 mt-1 cursor-pointer"
        id="theme-switch"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="w-8"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17.293 13.293A8 8 0 016.707 2.707a8.001 8.001 0 1010.586 10.586z"
          />
        </svg>
      </button>
      <a
        href="/organization-login"
        onClick={[Function]}
      >
        <button
          className="btn primary lg undefined"
          disabled={false}
          type="button"
        >
           
          Sign In
           
        </button>
      </a>
      <a
        href="/register-organization"
        onClick={[Function]}
      >
        <button
          className="btn transparentbtn lg mr-8"
          disabled={false}
          type="button"
        >
           
          Register an organization
           
        </button>
      </a>
    </div>
    <div
      className="flex px-5 lg:hidden"
    >
      <button
        className="px-3"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="w-6 mr-2"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17.293 13.293A8 8 0 016.707 2.707a8.001 8.001 0 1010.586 10.586z"
          />
        </svg>
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="w-7 dark:text-dark-text-fill"
          fill="none"
          stroke="currentColor"
          strokeWidth={2}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 6h16M4 12h16M4 18h16"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </button>
    </div>
  </div>
  <ul
    className="hidden"
  >
    <li
      className="p-2 w-full mt-2 dark:text-dark-text-fill text-primary"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="p-2 w-full dark:text-dark-text-fill"
    >
      <a
        href="/pricing"
        onClick={[Function]}
      >
        Pricing
      </a>
    </li>
    <li
      className="p-2 w-full dark:text-dark-text-fill"
    >
      <a
        className="w-full"
        href="/product"
        onClick={[Function]}
      >
        Product
      </a>
    </li>
    <li
      className="p-2 w-full dark:text-dark-text-fill mt-6 mb-2 bg-primary text-white rounded-md px-[35%]"
    >
      <a
        className="w-full"
        href="/organization-login"
        onClick={[Function]}
      >
         
        Sign In
      </a>
    </li>
    <a
      href="/register-organization"
      onClick={[Function]}
    >
      <button
        className="btn transparentbtn lg mr-8"
        disabled={false}
        type="button"
      >
         
        Register an organization
         
      </button>
    </a>
  </ul>
</div>
`;

exports[`Header Tests Should render the header with pricing active 1`] = `
<div
  className="w-screen h-[8vh] z-10 bg-white dark:bg-dark-bg fixed border-b undefined"
>
  <div
    className="px-3 flex justify-between items-center w-full h-full"
  >
    <div
      className="flex items-center h-full justify-between lg:w-full"
    >
      <a
        className="flex flex-row lg:px-5"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo"
          className="w-full cursor-pointer mr-2"
          src={Object {}}
        />
        <h1
          className="text-3xl font-bold font-lexend text-primary dark:text-dark-text-fill"
        >
          PULSE
        </h1>
      </a>
      <ul
        className="hidden lg:flex cursor-pointer"
      >
        <li
          className="px-5 text-xl dark:text-dark-text-fill"
        >
          <a
            className=""
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="px-5 text-xl dark:text-dark-text-fill"
        >
          <a
            aria-current="page"
            className="text-primary"
            href="/pricing"
            onClick={[Function]}
          >
            Pricing
          </a>
        </li>
        <li
          className="px-5 text-xl dark:text-dark-text-fill"
        >
          <a
            className=""
            href="/product"
            onClick={[Function]}
          >
            Product
          </a>
        </li>
      </ul>
    </div>
    <div
      className="hidden lg:flex lg:w-full justify-end "
    >
      <button
        className="px-4 mt-1 cursor-pointer"
        id="theme-switch"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="w-8"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17.293 13.293A8 8 0 016.707 2.707a8.001 8.001 0 1010.586 10.586z"
          />
        </svg>
      </button>
      <a
        href="/organization-login"
        onClick={[Function]}
      >
        <button
          className="btn primary lg undefined"
          disabled={false}
          type="button"
        >
           
          Sign In
           
        </button>
      </a>
      <a
        href="/register-organization"
        onClick={[Function]}
      >
        <button
          className="btn transparentbtn lg mr-8"
          disabled={false}
          type="button"
        >
           
          Register an organization
           
        </button>
      </a>
    </div>
    <div
      className="flex px-5 lg:hidden"
    >
      <button
        className="px-3"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="w-6 mr-2"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17.293 13.293A8 8 0 016.707 2.707a8.001 8.001 0 1010.586 10.586z"
          />
        </svg>
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="w-7 dark:text-dark-text-fill"
          fill="none"
          stroke="currentColor"
          strokeWidth={2}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 6h16M4 12h16M4 18h16"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </button>
    </div>
  </div>
  <ul
    className="hidden"
  >
    <li
      className="p-2 w-full mt-2 dark:text-dark-text-fill text-primary"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="p-2 w-full dark:text-dark-text-fill"
    >
      <a
        href="/pricing"
        onClick={[Function]}
      >
        Pricing
      </a>
    </li>
    <li
      className="p-2 w-full dark:text-dark-text-fill"
    >
      <a
        className="w-full"
        href="/product"
        onClick={[Function]}
      >
        Product
      </a>
    </li>
    <li
      className="p-2 w-full dark:text-dark-text-fill mt-6 mb-2 bg-primary text-white rounded-md px-[35%]"
    >
      <a
        className="w-full"
        href="/organization-login"
        onClick={[Function]}
      >
         
        Sign In
      </a>
    </li>
    <a
      href="/register-organization"
      onClick={[Function]}
    >
      <button
        className="btn transparentbtn lg mr-8"
        disabled={false}
        type="button"
      >
         
        Register an organization
         
      </button>
    </a>
  </ul>
</div>
`;

exports[`Header Tests Should render the header with product active 1`] = `
<div
  className="w-screen h-[8vh] z-10 bg-white dark:bg-dark-bg fixed border-b undefined"
>
  <div
    className="px-3 flex justify-between items-center w-full h-full"
  >
    <div
      className="flex items-center h-full justify-between lg:w-full"
    >
      <a
        className="flex flex-row lg:px-5"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo"
          className="w-full cursor-pointer mr-2"
          src={Object {}}
        />
        <h1
          className="text-3xl font-bold font-lexend text-primary dark:text-dark-text-fill"
        >
          PULSE
        </h1>
      </a>
      <ul
        className="hidden lg:flex cursor-pointer"
      >
        <li
          className="px-5 text-xl dark:text-dark-text-fill"
        >
          <a
            className=""
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="px-5 text-xl dark:text-dark-text-fill"
        >
          <a
            className=""
            href="/pricing"
            onClick={[Function]}
          >
            Pricing
          </a>
        </li>
        <li
          className="px-5 text-xl dark:text-dark-text-fill"
        >
          <a
            aria-current="page"
            className="text-primary"
            href="/product"
            onClick={[Function]}
          >
            Product
          </a>
        </li>
      </ul>
    </div>
    <div
      className="hidden lg:flex lg:w-full justify-end "
    >
      <button
        className="px-4 mt-1 cursor-pointer"
        id="theme-switch"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="w-8"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17.293 13.293A8 8 0 016.707 2.707a8.001 8.001 0 1010.586 10.586z"
          />
        </svg>
      </button>
      <a
        href="/organization-login"
        onClick={[Function]}
      >
        <button
          className="btn primary lg undefined"
          disabled={false}
          type="button"
        >
           
          Sign In
           
        </button>
      </a>
      <a
        href="/register-organization"
        onClick={[Function]}
      >
        <button
          className="btn transparentbtn lg mr-8"
          disabled={false}
          type="button"
        >
           
          Register an organization
           
        </button>
      </a>
    </div>
    <div
      className="flex px-5 lg:hidden"
    >
      <button
        className="px-3"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="w-6 mr-2"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17.293 13.293A8 8 0 016.707 2.707a8.001 8.001 0 1010.586 10.586z"
          />
        </svg>
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="w-7 dark:text-dark-text-fill"
          fill="none"
          stroke="currentColor"
          strokeWidth={2}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 6h16M4 12h16M4 18h16"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </button>
    </div>
  </div>
  <ul
    className="hidden"
  >
    <li
      className="p-2 w-full mt-2 dark:text-dark-text-fill text-primary"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="p-2 w-full dark:text-dark-text-fill"
    >
      <a
        href="/pricing"
        onClick={[Function]}
      >
        Pricing
      </a>
    </li>
    <li
      className="p-2 w-full dark:text-dark-text-fill"
    >
      <a
        className="w-full"
        href="/product"
        onClick={[Function]}
      >
        Product
      </a>
    </li>
    <li
      className="p-2 w-full dark:text-dark-text-fill mt-6 mb-2 bg-primary text-white rounded-md px-[35%]"
    >
      <a
        className="w-full"
        href="/organization-login"
        onClick={[Function]}
      >
         
        Sign In
      </a>
    </li>
    <a
      href="/register-organization"
      onClick={[Function]}
    >
      <button
        className="btn transparentbtn lg mr-8"
        disabled={false}
        type="button"
      >
         
        Register an organization
         
      </button>
    </a>
  </ul>
</div>
`;
